src/inc/quic_datapath.h

```diff
@@ -161,7 +161,15 @@ typedef struct CXPLAT_ROUTE {
     uint8_t NextHopLinkLayerAddress[6];
     void* Queue;
 
-    CXPLAT_ROUTE_STATE State; // Keep this as the last property in the struct.
+    //
+    // QuicCopyRouteInfo copies memory up to this point (not including State).
+    //
+
+    CXPLAT_ROUTE_STATE State;
+    //
+    // Indicates the queue has been updated for this route.
+    //
+    uint8_t QueueUpdated : 1;
```

Comment:
Why do we need a flag and not just treat a null `Queue`?

Reply:
I was trying to imitate PartitionUpdate so that queue only gets updated once after route is resolved.

Reply:
Do we need the queue to be non-null between the time that the route is set to RouteSuspected and the time that the new queue is assigned? If not then I also think just setting the queue to null would be better than adding a flag.

Reply:
Are you suggesting this?
```C
if (DatagramPath->Route.State == RouteResolved && !DatagramPath->Route.Queue) {
    DatagramPath->Route.Queue = Datagram->Queue;
}
```

Reply:
Yes, but please first answer my question in the other thread because it looks to me like the queue would normally already be reassigned in CxPlatResolveRoute at this point.